src/store/datasets-dicom.ts

```diff
@@ -117,104 +197,109 @@ export const useDICOMStore = defineStore('dicom', {
-              if (file) {
-                volumeFiles.push(file);
-              }
-            });
-
-            // Add file entries to the file store
-            fileStore.add(volumeKey, volumeFiles);
-
-            // Now read the tags
-            const numberOfSlices = volumeFiles.length;
-
-            if (!(volumeKey in this.volumeInfo)) {
-              const info = await dicomIO.readTags(volumeFiles[0], [
-                { name: 'PatientName', tag: '0010|0010', strconv: true },
-                { name: 'PatientID', tag: '0010|0020', strconv: true },
-                { name: 'PatientBirthDate', tag: '0010|0030' },
-                { name: 'PatientSex', tag: '0010|0040' },
-                { name: 'StudyInstanceUID', tag: '0020|000d' },
-                { name: 'StudyDate', tag: '0008|0020' },
-                { name: 'StudyTime', tag: '0008|0030' },
-                { name: 'StudyID', tag: '0020|0010', strconv: true },
-                { name: 'AccessionNumber', tag: '0008|0050' },
-                { name: 'StudyDescription', tag: '0008|1030', strconv: true },
-                { name: 'Modality', tag: '0008|0060' },
-                { name: 'SeriesInstanceUID', tag: '0020|000e' },
-                { name: 'SeriesNumber', tag: '0020|0011' },
-                { name: 'SeriesDescription', tag: '0008|103e', strconv: true },
-              ]);
-
-              // TODO parse the raw string values
-              const patient = {
-                PatientID: info.PatientID || ANONYMOUS_PATIENT_ID,
-                PatientName: info.PatientName || ANONYMOUS_PATIENT,
-                PatientBirthDate: info.PatientBirthDate || '',
-                PatientSex: info.PatientSex || '',
-              };
-              const patientKey = genSynPatientKey(patient);
-
-              const studyKey = info.StudyInstanceUID;
-              const study = pick(
-                info,
-                'StudyID',
-                'StudyInstanceUID',
-                'StudyDate',
-                'StudyTime',
-                'AccessionNumber',
-                'StudyDescription'
-              );
-
-              const volumeInfo = {
-                ...pick(
-                  info,
-                  'Modality',
-                  'SeriesInstanceUID',
-                  'SeriesNumber',
-                  'SeriesDescription'
-                ),
-                NumberOfSlices: numberOfSlices,
-                VolumeID: volumeKey,
-              };
+      volumesGrouped.forEach(async (layers) => {
+        const { volumeKey, tags } = layers[0];
+
+        const makeVolumeInfo = ({
+          tags: layerTags,
+          files,
+        }: VolumeIngredients) => ({
+          ...pick(
+            layerTags,
+            'Modality',
+            'SeriesInstanceUID',
+            'SeriesNumber',
+            'SeriesDescription'
+          ),
+          NumberOfSlices: files.length,
+          VolumeID: volumeKey,
+          files,
+        });
```

Comment:
Volumes in the dicomStore hold references to files to group files by layer.  Is that bad?  Do we need the fileStore?

Reply:
I think the volumes shouldn't be concerned with layers for now. Layers should be handled at the scene level. That way, volumes can remain simple.